backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -710,52 +683,45 @@ public UserInfo getUserByLoginEmail(String loginEmail) {
       }
 
-      return consolidateUser(
-          apiUser,
-          oktaUser.getOrganizationRoleClaims(),
-          oktaUser.getStatus(),
-          oktaUser.isSiteAdmin());
+      return consolidateUser(apiUser, oktaUser);
     } catch (IllegalGraphqlArgumentException | UnidentifiedUserException e) {
       throw new OktaAccountUserException();
     }
   }
 
-  private UserInfo consolidateUser(
-      ApiUser apiUser,
-      Optional<OrganizationRoleClaims> optClaims,
-      UserStatus userStatus,
-      boolean isSiteAdmin) {
-
-    OrganizationRoleClaims claims = optClaims.orElseThrow(UnidentifiedUserException::new);
-
-    // use the target user's org so response is built correctly even if site admin is the requester
-    Organization org = _orgService.getOrganization(claims.getOrganizationExternalId());
-
-    List<Facility> facilities = _orgService.getFacilities(org);
-    Set<Facility> facilitiesSet = new HashSet<>(facilities);
-
-    boolean allFacilityAccess = claims.grantsAllFacilityAccess();
-    Set<Facility> accessibleFacilities =
-        allFacilityAccess
-            ? facilitiesSet
-            : facilities.stream()
-                .filter(f -> claims.getFacilities().contains(f.getInternalId()))
-                .collect(Collectors.toSet());
-
-    OrganizationRoles orgRoles =
-        new OrganizationRoles(org, accessibleFacilities, claims.getGrantedRoles());
-    if (!_featureFlagsConfig.isOktaMigrationEnabled() && !isSiteAdmin) {
-      try {
-        setRolesAndFacilities(orgRoles, apiUser);
-      } catch (PrivilegeUpdateFacilityAccessException e) {
-        log.warn(
-            "Could not migrate roles and facilities for user with id={} because facilities were invalid",
-            apiUser.getInternalId());
+  private UserInfo consolidateUser(ApiUser apiUser, PartialOktaUser oktaUser) {
+    boolean isSiteAdmin = oktaUser.isSiteAdmin();
+    UserStatus userStatus = oktaUser.getStatus();
+    OrganizationRoleClaims oktaClaims =
+        oktaUser.getOrganizationRoleClaims().orElseThrow(UnidentifiedUserException::new);
+    OrganizationRoles orgRoles = _orgService.getOrganizationRoles(oktaClaims);
+
+    if (!isSiteAdmin) {
+      if (_featureFlagsConfig.isOktaMigrationEnabled()) {
+        orgRoles = getOrgRolesFromDB(apiUser);
+      } else {
+        setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
       }
     }
+
     return new UserInfo(apiUser, Optional.of(orgRoles), isSiteAdmin, userStatus);
   }
 
+  private UserInfo consolidateNonSiteAdminUser(
+      ApiUser apiUser, Optional<OrganizationRoleClaims> oktaClaims) {
+    OrganizationRoles orgRoles = null;
+    if (oktaClaims.isPresent()) {
+      orgRoles = _orgService.getOrganizationRoles(oktaClaims.get());
+    }
+
+    if (_featureFlagsConfig.isOktaMigrationEnabled()) {
+      orgRoles = getOrgRolesFromDB(apiUser);
+    } else {
+      setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
```

Comment:
It's a little strange to me that we have this set method `else`ed against the feature flag check for the migration, especially since `orgRoles` might be null. Could we potentially move it up after 714 to only save if we have a valid grab from Okta? That feels more in line with how I'm understanding our syncing procedure to be working.

Reply:
When the feature flag is enabled and Okta claims are present, we should not still be setting roles based on the Okta claims right? Since the feature flag being turned on means the DB should be source of truth? Also seems like `setRolesAndFacilities` would allow nullable `orgRoles`

Reply:
Yes, we should only be setting them in the DB if the feature flag is off.